src/main/java/taxi/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package taxi.service;
+
+import java.util.Optional;
+import javax.naming.AuthenticationException;
+import taxi.dao.DriverDao;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverDao driverDao;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverDao.findByLogin(login);
+        if (driver.isEmpty() || driver.get().getPassword().equals(password)) {
+            throw new AuthenticationException("Login or password doesn't correct");
+        }
```

Comment:
Try login with the correct login and incorrect password.
And try with correct login and password)
It works incorrect